### PR TITLE
Api-000-med-dose-and-rate

### PR DESCRIPTION
**Pull request by msilverman-va**

### CODE REVIEW

File: vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/medication/R4MedicationTransformers.java [hunk 1]
```diff
@@ -4,30 +4,56 @@
...
+
     return Dosage.builder()
         .text(trimToNull(sig))
         .patientInstruction(trimToNull(ptInstructions))
+        .doseAndRate(doseAndRate(doseValue, doseUnit))
         .build();
   }
 
+  private static List<DoseAndRate> doseAndRate(String doseValue, String doseUnit) {
+    if (NumberUtils.isParsable(doseValue)) {
```
Comment (jhulbert-va) [inline vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/medication/R4MedicationTransformers.java:44]: Why not do our own try catch here?

File: vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/medication/R4MedicationTransformers.java [hunk 2]
```diff
@@ -4,30 +4,56 @@
...
+      return List.of(
+          DoseAndRate.builder()
+              .doseQuantity(
+                  SimpleQuantity.builder()
+                      .unit(trim(doseUnit))
+                      .value(BigDecimal.valueOf(Double.parseDouble(doseValue)))
```
Comment (jhulbert-va) [inline vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/medication/R4MedicationTransformers.java:50]: I think just parse the string to a BigDecimal. Parsing to a Double first is redundant.

Comment (bschofield-va) [reply vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/medication/R4MedicationTransformers.java:50]: `R4Transformers.toBigDecimal(...)`

File: vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/medication/R4MedicationTransformers.java (hunk 1 above)
Comment (jnmiller-va) [reply vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/medication/R4MedicationTransformers.java:44]: There won't be an error if we use `R4Transformers.toBigDecimal`. How would you like to proceed?

Comment (jnmiller-va) [reply vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/medication/R4MedicationTransformers.java:44]: Apologies. There will be a NPE on input of `1 CAPSULE`.

Comment (jnmiller-va) [reply vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/medication/R4MedicationTransformers.java:44]: Noticing `toBigDecimal` doesn't allow `.4` or `0.4`.

Comment (bschofield-va) [reply vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/medication/R4MedicationTransformers.java:44]: I think the implementation of `toBigDecimal(..)` is incorrect. Probably should be ...
```
  public static BigDecimal toBigDecimal(String maybeDecimal) {
    if (isBlank(maybeDecimal)) {
      return null;
    }
    try {
      return new BigDecimal(maybeDecimal);
    } catch (NumberFormatException e) {
      return null;
    }
  }
```

Comment (jnmiller-va) [reply vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/medication/R4MedicationTransformers.java:44]: Awesome! I appreciate the quick feedback!

File: vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/medication/R4MedicationTransformers.java [hunk 3]
```diff
@@ -1,30 +1,59 @@
...
+
     return Dosage.builder()
         .text(trimToNull(sig))
         .patientInstruction(trimToNull(ptInstructions))
+        .doseAndRate(doseAndRates.isEmpty() ? null : doseAndRates)
+        .build();
+  }
+
+  private static DoseAndRate doseAndRate(Dose dose) {
+    String unit = trim(dose.units());
```
Comment (bschofield-va) [inline vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/medication/R4MedicationTransformers.java:50]: I think `trimToNull` it preferrable

**Pull request merged**